balloc/balloc.c

```diff
@@ -3045,19 +3096,29 @@ M0_INTERNAL int m0_balloc_create(uint64_t              cid,
 			rc = -ENOMEM;
 		} else {
 			cb->cb_container_id = cid;
+			cb->group_count     = group_count;
+                        M0_BE_ALLOC_CREDIT_ARR(cb->cb_db_group_extents,
+                                                cb->group_count, seg, &cred);
+                        M0_BE_ALLOC_ARR_SYNC(cb->cb_db_group_extents,
+                                                cb->group_count, seg, &tx);
+                        if(cb->cb_db_group_extents == NULL)
+                                return M0_RC(rc);
 
 			balloc_format_init(cb);
-			m0_be_btree_init(&cb->cb_db_group_extents, seg,
-					 &ge_btree_ops);
+			for (i = 0; i < cb->group_count; i++) {
+				m0_be_btree_init(&cb->cb_db_group_extents[i], seg,
+						 &ge_btree_ops);
+			}
 			m0_be_btree_init(&cb->cb_db_group_desc, seg,
 					 &gd_btree_ops);
 			rc = balloc_trees_create(cb, &tx, fid);
 			if (rc == 0) {
 				M0_BE_TX_CAPTURE_PTR(seg, &tx, cb);
 				*out = cb;
 			} else {
-				m0_be_btree_fini(&cb->cb_db_group_extents);
-				m0_be_btree_fini(&cb->cb_db_group_desc);
+				for (k = 0; k < i; k++) {
```

Comment:
No need to define another variable 'k' here.
Because 'i' equals "cb->group_count" at this moment.
Just use 'i' here.

Reply:
What about "cb->cb_db_group_desc" btree?

Reply:
During balloc_trees_create() fail inside loop,   i may not be cb->group_count, So we need to revert all previous created btree,  here   0<= i  < cb->group_count
cb->cb_db_group_desc  will be created latter, on this point cb->cb_db_group_desc" has not been created.

Probabelly we need to handle independently cb->cb_db_group_desc btree creation failure.  This code is currently not present. may ben I can add new code for error handling  of  cb->cb_db_group_desc fail " to destruct all btrees".